x-pack/plugins/security_solution/public/detection_engine/rule_management_ui/components/rules_table/rules_table/use_rules_table_saved_state.ts

```diff
@@ -41,61 +37,55 @@ function validateState(
 ): [RulesTableSavedFilter, RulesTableSavedSorting, RulesTableUrlSavedPagination] {
   const [filterFromUrl] = validateNonExact(urlState, RulesTableSavedFilter);
   const [filterFromStorage] = validateNonExact(storageState, RulesTableSavedFilter);
-  const filter = { ...filterFromStorage, ...filterFromUrl };
+  const filter: RulesTableSavedFilter = {
+    searchTerm: filterFromUrl?.searchTerm ?? filterFromStorage?.searchTerm,
+    source: filterFromUrl?.source ?? filterFromStorage?.source,
+    tags: filterFromUrl?.tags ?? filterFromStorage?.tags,
+    enabled: filterFromUrl?.enabled ?? filterFromStorage?.enabled,
+  };
 
   const [sortingFromUrl] = validateNonExact(urlState, RulesTableSavedSorting);
   const [sortingFromStorage] = validateNonExact(storageState, RulesTableSavedSorting);
-  const sorting = { ...sortingFromStorage, ...sortingFromUrl };
+  const sorting = {
+    field: sortingFromUrl?.field ?? sortingFromStorage?.field,
+    order: sortingFromUrl?.order ?? sortingFromStorage?.order,
+  };
```

Comment:
Can we roll back to the previous implementation which looks simpler?

```ts
const filter = { ...filterFromStorage, ...filterFromUrl };
const sorting = { ...sortingFromStorage, ...sortingFromUrl };
```

Reply:
I've added an explanation comment